src/integration_tests/helpers/integration_helpers.py

```diff
@@ -0,0 +1,284 @@
+            return
+        else:
+            attempts -= 1
+            time.sleep(backoff)
+            backoff += backoff
+
+
+def cleanup() -> None:
+    """
+    Method to cleanup all test data created depending on local or remote run.
+    Should be ran before and after test to account for test failures.
+
+    Parameters:
+        None
+
+    Returns:
+        None
+    """
+    if config.API_URL.__contains__("local"):
+        _delete_local_firestore_data()
+
+        _delete_local_bucket_data("devtools/gcp-storage-emulator/data/schema_bucket/")
+
+        _delete_local_bucket_data("devtools/gcp-storage-emulator/data/dataset_bucket/")
+    else:
+        _delete_blobs(storage_client.get_bucket(config.DATASET_BUCKET_NAME))
+
+        _delete_blobs(storage_client.get_bucket(config.SCHEMA_BUCKET_NAME))
+
+        _delete_collection(db.collection("datasets"))
+
+        _delete_collection(db.collection("schemas"))
+
+
+def _delete_local_firestore_data():
+    """
+    Method to cleanup local test data in the emulated firestore instance.
+
+    Parameters:
+        None
+
+    Returns:
+        None
+    """
+    requests.delete(
+        "http://localhost:8080/emulator/v1/projects/mock-project-id/databases/(default)/documents"
+    )
+
+
+def _delete_local_bucket_data(filepath: str):
+    """
+    Method to cleanup local test data in the bucket instance.
+
+    Parameters:
+        filepath: the filepath for the bucket instance to be deleted
+
+    Returns:
+        None
+    """
+    path_instance = Path(filepath)
+    if Path.is_dir(path_instance):
+        shutil.rmtree(path_instance)
+
+
+def _delete_blobs(bucket) -> None:
+    """
+    Method to delete all blobs in the specified bucket.
+
+    Parameters:
+        bucket: the bucket to clean
+
+    Returns:
+        None
+    """
+    blobs = bucket.list_blobs()
+
+    for blob in blobs:
+        blob.delete()
+
+
+def _delete_collection(collection_ref: firestore.CollectionReference) -> None:
+    """
+    Recursively deletes the collection and its subcollections.
+    Parameters:
+    collection_ref (firestore.CollectionReference): the reference of the collection being deleted.
+    """
+    doc_collection = collection_ref.stream()
+
+    for doc in doc_collection:
+        _recursively_delete_document_and_sub_collections(doc.reference)
+
+
+def _recursively_delete_document_and_sub_collections(
+    doc_ref: firestore.DocumentReference,
+) -> None:
+    """
+    Loops through each collection in a document and deletes the collection.
+    Parameters:
+    doc_ref (firestore.DocumentReference): the reference of the document being deleted.
+    """
+    for collection_ref in doc_ref.collections():
+        _delete_collection(collection_ref)
+
+    doc_ref.delete()
+
+
+MOCK_UNIT_RESPONSE = {
+    "schema_version": "v1.0.0",
+    "sds_schema_version": 4,
+    "survey_id": "test_survey_id",
+    "period_id": "abc",
+    "data": "<encrypted data>",
+}
```

Comment:
Is it better if we put all the data, either the data that run through the test or to validate the test, into test data?

Reply:
Will move